sway-lib-std/src/chain/auth.sw

```diff
@@ -47,34 +47,42 @@ pub fn msg_sender() -> Result<Identity, AuthError> {
 /// Get the owner of the inputs (of type `InputCoin` or `InputMessage`) to a
 /// TransactionScript if they all share the same owner.
 fn inputs_owner() -> Result<Identity, AuthError> {
-    let inputs_count = tx_inputs_count();
+    let inputs = input_count();
     let mut candidate = Option::None::<Address>();
-    let mut i = 0;
+    let mut i = 0u8;
 
     // Note: `inputs_count` is guaranteed to be at least 1 for any valid tx.
-    while i < inputs_count {
-        let input_type = tx_input_type(i);
-        if input_type != INPUT_COIN && input_type != INPUT_MESSAGE {
-            // type != InputCoin or InputMessage, continue looping.
-            i += 1;
-            continue;
+    while i < inputs {
+        let type_of_input = input_type(i);
+        match type_of_input {
+            Input::Coin => {
+                ();
+            },
+            Input::Message => {
+                ();
+            },
+            _ => {
+                // type != InputCoin or InputMessage, continue looping.
+                i += 1u8;
+                continue;
+            }
         }
```

Comment:
Does the `_` check for multiple types? Would it be cleaner to use an `if` expression instead? In `inputs.sw` it looks like it looks for a single input, a Contract. I think this should use an `if` expression since that would halve the number of lines of code

Reply:
The code is skipping inputs which aren't `Coin` or `Message` by continuing the loop, as the comment says.  The whole thing is a bit fragile, but the best alternative is iterators.

Reply:
As per Toby's comment, leaving this as is for now.